raft/server.cc

```diff
@@ -1173,8 +1196,11 @@ future<> server_impl::abort() {
         f.set_exception(std::runtime_error("Snapshot application aborted"));
     }
 
-    co_await seastar::when_all_succeed(std::move(abort_rpc), std::move(abort_sm), std::move(abort_persistence)).discard_result();
-
+    // Destroy entry waiters before waiting for `abort_rpc`,
+    // since the RPC implementation may wait for forwarded `modify_config` calls to finish
+    // (and `modify_config` does not finish until the configuration entry is committed or an error occurs).
+    // FIXME: probably need to do the same with read barriers (`_reads`)
```

Comment:
Please resolve this fixme. Going forward it will be confusing. It's fine to abort read barriers with an exception, let's just do it.

Reply:
It's fixed in https://github.com/scylladb/scylla/pull/10561